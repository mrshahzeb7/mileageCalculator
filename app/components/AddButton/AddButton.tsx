import React from 'react';
import {StyleSheet} from 'react-native';
import {icons} from '../../assets';
import {colors, iconSizes} from '../../theme';
import {responsiveHeight, responsiveWidth} from '../../utils';
import {Icon} from '../Icon/Icon';

interface Props {
  onPress?: () => void;
}
const size = iconSizes.extralarge;
export const AddButton: React.FC<Props> = ({onPress}) => {
  return (
    <Icon
      onPress={onPress}
      size={iconSizes.small}
      containerStyle={styles.container}
      icon={icons.Plus}
    />
  );
};

// styles
const styles = StyleSheet.create({
  container: {
    height: size,
    width: size,
    borderRadius: size / 2,
    backgroundColor: colors.primaryButton,
    alignItems: 'center',
    justifyContent: 'center',
    position: 'absolute',
    zIndex: 100,
    bottom: responsiveHeight(13),
    right: responsiveWidth(5),
  },
});
