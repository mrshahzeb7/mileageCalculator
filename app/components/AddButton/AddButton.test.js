import 'react-native';
import React from 'react';
import {AddButton} from './AddButton';
import renderer from 'react-test-renderer';

describe('AddButton', () => {
  it(' AddButton renders correctly', () => {
    const tree = renderer.create(<AddButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
