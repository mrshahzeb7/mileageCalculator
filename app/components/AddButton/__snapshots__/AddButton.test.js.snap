// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddButton  AddButton renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#93b2fd",
      "borderRadius": 50.02435032061885,
      "bottom": 173.42000000000002,
      "height": 100.0487006412377,
      "justifyContent": "center",
      "position": "absolute",
      "right": 37.5,
      "width": 100.0487006412377,
      "zIndex": 100,
    }
  }
>
  <Image
    source={
      {
        "testUri": "../../../app/assets/icons/general/plus.png",
      }
    }
    style={
      [
        {
          "resizeMode": "contain",
        },
        null,
        {
          "height": 50.02435032061885,
          "width": 50.02435032061885,
        },
        undefined,
      ]
    }
  />
</View>
`;
